### PR TITLE
EVG-6267 periodic builds - configuration

### PR DESCRIPTION
**Pull request by john-m-liu**
Screen Shot 2019-06-26 at 10 01 55 AM

### CODE REVIEW

File: model/project_ref.go [hunk 1]
```diff
@@ -5,6 +5,8 @@ import (
 	"math"
 	"regexp"
 
+	"github.com/mongodb/grip"
+
```
Comment (tychoish) [inline model/project_ref.go]: omit blank line

File: model/project_ref.go [hunk 2]
```diff
@@ -510,6 +522,21 @@ func (t TriggerDefinition) Validate(parentProject string) error {
 	return nil
 }
 
+func (d PeriodicBuildDefinition) Validate() error {
```
Comment (tychoish) [inline model/project_ref.go]: should this be a pointer receiver?

File: model/project_ref.go [hunk 3]
```diff
@@ -510,6 +522,21 @@ func (t TriggerDefinition) Validate(parentProject string) error {
...
+	catcher := grip.NewBasicCatcher()
+	if d.IntervalHours <= 0 {
+		catcher.Add(errors.New("Interval must be a positive integer"))
```
Comment (tychoish) [inline model/project_ref.go]: I think you can just catcher.New()

File: service/project.go [hunk 4]
```diff
@@ -432,6 +434,10 @@ func (uis *UIServer) modifyProject(w http.ResponseWriter, r *http.Request) {
 		catcher.Add(event.RemoveSubscription(id))
 	}
 
+	for i, buildDef := range responseRef.PeriodicBuilds {
+		catcher.Add(errors.Wrapf(buildDef.Validate(), "invalid periodic build definition on line %d", i+1))
```
Comment (tychoish) [inline service/project.go]: catcher.Wrapf()!

File: model/project_ref.go (hunk 1 above)
Comment (john-m-liu) [reply model/project_ref.go]: fixed

File: model/project_ref.go (hunk 2 above)
Comment (john-m-liu) [reply model/project_ref.go]: Don't think it matters here but I changed it

File: model/project_ref.go (hunk 3 above)
Comment (john-m-liu) [reply model/project_ref.go]: fixed

File: service/project.go (hunk 4 above)
Comment (john-m-liu) [reply service/project.go]: changed

**Pull request merged**